capsules/src/screen.rs

```diff
@@ -162,7 +152,7 @@ impl<'a> Screen<'a> {
         {
             Err(e) => CommandReturn::failure(e),
             Ok(r) => {
-                if self.screen_ready.get() && self.current_process.is_none() {
+                if self.current_process.is_none() {
```

Comment:
That was my goal. Preventing the call in the syscall driver would be redundant.

Reply:
I might be missing something, but it seems to me from the code that you forward the application's command right away if there is no other process that runs a command.

Reply:
Yes, and that should return not BUSY, but rather OFF, because of the HIL (and I should add that to the HIL). That's also what the userspace should see, so there's no problem just forwarding the calls without interference.

Reply:
Yes, OFF is better.

I would go for one of these:
1. schedule an upcall as soon as the display is ready so that the application does not need to poll that
    - this might be tricky, as an application that registers for this upcall after the screen has sent the ready notification should receive an upcall immediately upon registration
2. queue the command until the screen becomes ready (what is being done now)

Reply:
I don't envision it as the application needs to busy wait. The HIL mandates that every display on/off call will result in a callback, and that's trivial to forward (and this seems to be already working in my testing).

To prevent the situation where an application registers upcall on a screen that's already turned on, the application should turn it on again. That's also in the HIL, in the same place: every display on/off call shall result in a callback.

Reply:
> I don't envision it as the application needs to busy wait. The HIL mandates that every display on/off call will result in a callback, and that's trivial to forward (and this seems to be already working in my testing).
> 
The initialization of the screen is different from the power on. At least the screens that I used need to perform an initialization sequence before they can even be powered on.

Reply:
What's the difference between power on and initialization?

Reply:
My understanding from reading the screen's documentation is that power on and off are for powering the screen LCD/TFT, while the initialization sequence is for the screen controller.

Reply:
Is there a case where they need to be done separately? If not, then no problem: they both get started on set_power.

If yes, then set_power could initialize the controller, and set_brightness could power on the panel.

Reply:
The display that I am using has a *need to reset* before it can be powered up. This reset is in the hunderds of ms
https://github.com/tock/tock/blob/35cfc73e2b70023a91bbd550b6416366c3ca4911/capsules/src/st77xx.rs#L572-L588

These seem to be necessary to reset the controller and should not for power on and off. If this time is acceptable, we could merge the reset and power on into a single command.

Reply:
The screens that I used do not have the ability to be fully powered off, I think only the TFT is actually powered off, while the internal memory and controller still use power.

There are two ways to go now:
1. include this sequence in the power_on and delete the `screen_is_ready` event, with the observation that the power_on function will take a about a second to complete and the screen's memory will be fully erased;
2. keep this init sequence as it is right now, but then we do need this screen_is_ready function.

@hudson-ayers your feedback would be great

Reply:
I had a little bit more time to dig into this and I think @dcz-self you are right, we should drop the `screen_is_ready` event. Using `set_power` to turn off the screen (which may result in resetting the display's framebuffer memory) should be acceptable. Turning off only the display should be done by using `set_brightness` with parameter 0. 

With these changes this should be ready to go.

Reply:
I think there's a misunderstanding somewhere, because screen_is_ready is still there. I didn't do anything that would remove it, and indeed I rely on it for initialization. As for set_power and set_brightness, this is what I intended the documentation to read. Should I reword it in some way?

Reply:
I think it's ok.